Python/pywarpx/picmi.py

```diff
@@ -417,7 +419,11 @@ def initialize_inputs(self):
 
         # Maximum allowable size of each subdomain in the problem domain;
         #    this is used to decompose the domain for parallel calculations.
-        pywarpx.amr.max_grid_size = self.max_grid_size
+        if self.max_grid_size_x is None or self.max_grid_size_y is None:
+            pywarpx.amr.max_grid_size = self.max_grid_size
+        else:
+            pywarpx.amr.max_grid_size_x = self.max_grid_size_x
+            pywarpx.amr.max_grid_size_y = self.max_grid_size_y
         pywarpx.amr.blocking_factor = self.blocking_factor
```

Comment:
Can you also add the same input for the blocking factor, since it allows the same variants, `blocking_factor_[x,y,z]`?

Reply:
This should be done in https://github.com/ECP-WarpX/WarpX/pull/2524/commits/45c20ba67407c7312e05535b38c9bb22ba49b416.